R/exercise.R

```diff
@@ -248,18 +250,59 @@ evaluate_exercise <- function(exercise, envir, evaluate_global_setup = FALSE) {
   knitr::opts_chunk$set(comment = NA)
   knitr::opts_chunk$set(error = FALSE)
 
+  # helper function that unpacks knitr chunk options and
+  # returns a single character vector (e.g. "tidy=TRUE, prompt=FALSE")
+  unpack_options <- function(opts) {
+    # filter out options we don't need for the exercise.Rmd
+    opts <- opts[!(names(opts) %in% c("label", "engine"))]
+    opts <- opts[!grepl("exercise", names(opts))]
+    if (length(opts) == 0)
+      return(NULL)
+    paste0(names(opts), '=', opts)
+  }
 
+  # helper function that processes a list of raw setup chunks and
+  # returns a single character vector of knitr chunks for an Rmd file
+  get_setup_chunk_rmds <- function(chunks) {
+    if (is.null(chunks)) return()
```

Comment:
Always return values if the response is to be used
```suggestion
    if (is.null(chunks)) return(NULL)
```